src/server/routes/backend-wallet/import.ts

```diff
@@ -1,29 +1,48 @@
-import { Static, Type } from "@sinclair/typebox";
-import { FastifyInstance } from "fastify";
+import { Type, type Static } from "@sinclair/typebox";
+import type { FastifyInstance } from "fastify";
 import { StatusCodes } from "http-status-codes";
-import { WalletType } from "../../../schema/wallet";
 import { getConfig } from "../../../utils/cache/getConfig";
 import { createCustomError } from "../../middleware/error";
 import { AddressSchema } from "../../schemas/address";
 import { standardResponseSchema } from "../../schemas/sharedApiSchemas";
+import { getGcpKmsResourcePath } from "../../utils/wallets/gcpKmsResourcePath";
 import { importAwsKmsWallet } from "../../utils/wallets/importAwsKmsWallet";
 import { importGcpKmsWallet } from "../../utils/wallets/importGcpKmsWallet";
 import { importLocalWallet } from "../../utils/wallets/importLocalWallet";
 
 const RequestBodySchema = Type.Union([
   Type.Object({
-    awsKmsKeyId: Type.String({
-      description: "AWS KMS key ID",
-      examples: ["12345678-1234-1234-1234-123456789012"],
-    }),
+    label: Type.Optional(
+      Type.String({
+        description: "Optional label for the imported wallet",
+      }),
+    ),
     awsKmsArn: Type.String({
       description: "AWS KMS key ARN",
       examples: [
         "arn:aws:kms:us-east-1:123456789012:key/12345678-1234-1234-1234-123456789012",
       ],
     }),
+    credentials: Type.Object(
+      {
+        awsAccessKeyId: Type.String({ description: "AWS Access Key ID" }),
+        awsSecretAccessKey: Type.String({
+          description: "AWS Secret Access Key",
+        }),
+      },
+      {
+        description:
+          "Optional AWS credentials to use for importing the wallet, if not provided, the default AWS credentials will be used (if available).",
+      },
+    ),
```

Comment:
As mentioned above, I'd rather only prompt the user for _metadata_ and not _credentials_